bigquery/google/cloud/bigquery/query.py

```diff
@@ -31,7 +31,6 @@ class _SyncQueryConfiguration(object):
     Values which are ``None`` -> server defaults.
     """
     _default_dataset = None
-    _dry_run = None
```

Comment:
Please back out changes to `Query` (which is for synchronous aka "non-job" queries):  the field is correctly handled here.  See https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs/query#dryRun